src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,508 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "../interfaces/IServiceManager.sol";
+import "../interfaces/IStakeRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "./StakeRegistryStorage.sol";
+
+/**
+ * @title Interface for a `Registry` that keeps track of stakes of operators for up to 256 quroums.
+ * @author Layr Labs, Inc.
+ */
+contract StakeRegistry is StakeRegistryStorage {
+    // EVENTS
+    /// @notice emitted whenever the stake of `operator` is updated
+    event StakeUpdate(
+        bytes32 operatorId,
```

Comment:
should this event topic be indexed?

Reply:
good call, could be useful: https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/fa1d1cf99e47ca16a6d5422be8ea794fe8f3a494